modules/getComponents.js

```diff
@@ -8,7 +9,13 @@ function getComponentsForRoute(nextState, route, callback) {
 
   const getComponent = route.getComponent || route.getComponents
   if (getComponent) {
-    getComponent.call(route, nextState, callback)
+    const componentReturn = getComponent.call(route, nextState, callback)
+    if (isPromise(componentReturn))
+      componentReturn
+        // Try module.default first in case of System.import and Babel 6
+        .then(component => component.default || component)
+        .then(component => callback(null, component))
+        .catch(callback)
```

Comment:
I think this should be

```
.then(
  component => callback(null, component),
  callback
)
```

We don't want to catch exceptions thrown in running the callback.